src/api/streams/updateStream.js

```diff
@@ -13,9 +13,13 @@ import { apiPatch } from '../apiFetch';
 export default (
   auth: Auth,
   id: number,
-  property: string,
-  value: string | boolean,
-): Promise<ApiResponse> =>
-  apiPatch(auth, `streams/${id}`, {
-    [property]: value,
-  });
+  params: $ReadOnly<{|
+    description?: string,
+    new_name?: string,
+    is_private?: boolean,
+    is_web_public?: boolean,
+    stream_post_policy?: number,
+    history_public_to_subscribers?: boolean,
+    message_retention_days?: number,
```

Comment:
From the API doc, this should be `number | 'realm_default' | 'unlimited'`. But in fact let's leave it out for now, for the same reasons as `stream_post_policy` above.

Reply:
It may end up being even more complicated than that; see [discussion](https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/message_retention_days/near/1367895).